oak-blob-cloud/src/main/java/org/apache/jackrabbit/oak/blob/cloud/s3/S3Backend.java

```diff
@@ -939,10 +941,26 @@ private URI createPresignedURI(DataIdentifier identifier,
             final Date expiration = new Date();
             expiration.setTime(expiration.getTime() + expirySeconds * 1000);
 
-            GeneratePresignedUrlRequest request = new GeneratePresignedUrlRequest(bucket, key)
-                    .withMethod(method)
-                    .withExpiration(expiration);
-
+            GeneratePresignedUrlRequest request = null;
+            if (properties.getProperty(S3Constants.S3_ENCRYPTION)
```

Comment:
`properties.getProperty(S3Constants.S3_ENCRYPTION)` if null will lead to NPE. There should be a check. Also, if not already caught that means relevant unit test missing.